assets/src/apps/delivery/layouts/deck/DeckLayoutView.tsx

```diff
@@ -42,13 +43,40 @@ const InjectedStyles: React.FC<{ css?: string }> = (props) => {
   return injected ? <style>{injected}</style> : null;
 };
 
+const getActualOwnerDetails = (expression: string, sequence: any) => {
+  const [activityId, part] = expression.split('|');
+  const activitySequence = sequence.find((entry: any) => entry?.custom?.sequenceId === activityId);
+  const activityLayerId = activitySequence?.custom?.layerRef;
+  let expressionValue = getValue(expression, defaultGlobalEnv);
+  let updatedExpression = expression;
+  if (expressionValue == undefined) {
+    expressionValue = getValue(activityLayerId + '|' + part, defaultGlobalEnv);
+    if (expressionValue !== undefined) {
+      updatedExpression = activityLayerId + '|' + part;
+    } else if (activityLayerId) {
+      updatedExpression = activityLayerId + '|' + part;
+      updatedExpression = getActualOwnerDetails(updatedExpression, sequence);
+    }
+  }
+  return updatedExpression;
+};
+
+export const getPartOwnerVariableId = (expression: string, sequence: any) => {
+  if (expression.indexOf('stage.') === -1) {
+    return { expression, newExpression: expression };
+  }
+  const updatedExpression = getActualOwnerDetails(expression, sequence);
+  return { expression, newExpression: updatedExpression };
```

Comment:
the name of this method confuses me a bit, `getPartOwnerVariableId` returns `{ expression: string, newExpression: string }` that doesn't seem like an id or variable id to me.